library/src/blas3/herk_syrk_device.hpp

```diff
@@ -1379,6 +1379,11 @@ rocblas_copy_triangular_excluding_diagonal_kernel(rocblas_int    n,
             }
         }
 
+        // When copying back to C, we need to zero-out diagonal
+        if constexpr(!copy_from_C_to_W_C && rocblas_is_complex<T>)
+            if(row == col)
+                C[row + row * int64_t(ldc)] = std::real(C[row + row * int64_t(ldc)]);
```

Comment:
comment questions.     zero-out diagonal imaginary ?  
kernel name is now a misnomer "rocblas_copy_triangular_excluding_diagonal_kernel"

Reply:
Do you like rocblas_copy_triangular_herk_syrk_kernel?